source/features/clean-conversation-headers.tsx

```diff
@@ -1,68 +1,86 @@
-import {observe} from 'selector-observer';
+import elementReady from 'element-ready';
+import {ArrowLeftIcon} from '@primer/octicons-react';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
+import getDefaultBranch from '../github-helpers/get-default-branch';
+import onConversationHeaderUpdate from '../github-events/on-conversation-header-update';
 
-const deinit: VoidFunction[] = [];
+async function initIssue(): Promise<void> {
+	const byline = await elementReady('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)');
+	if (!byline) {
+		return;
+	}
 
-function initIssue(): void {
-	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
-			byline.classList.add('rgh-clean-conversation-header');
-			const {childNodes: bylineNodes} = byline;
-			// Removes: octocat opened this issue on 1 Jan [·] 1 comments
-			bylineNodes[4].textContent = bylineNodes[4].textContent!.replace('·', '');
+	byline.classList.add('rgh-clean-conversation-header');
 
-			// Removes: [octocat opened this issue on 1 Jan] · 1 comments
-			for (const node of [...bylineNodes].slice(0, 4)) {
-				node.remove();
-			}
-		}
-	});
-	deinit.push(observer.abort);
+	// Removes: [octocat opened this issue on 1 Jan] · 1 comments
+	for (let i = 0; i < 4; i++) {
+		byline.firstChild!.remove();
+	}
+
+	// Removes: octocat opened this issue on 1 Jan [·] 1 comments
+	byline.firstChild!.textContent = byline.firstChild!.textContent!.replace('·', '');
 }
 
-function initPR(): void {
-	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
-			byline.classList.add('rgh-clean-conversation-header');
-			const isSameAuthor = select('.js-discussion > .TimelineItem:first-child .author')?.textContent === select('.author', byline)!.textContent;
-			const baseBranch = select('.commit-ref.base-ref', byline)!;
-			const isDefaultBranch = (baseBranch.firstElementChild as HTMLAnchorElement).pathname.split('/').length === 3;
+async function initPR(): Promise<void> {
+	const byline = await elementReady('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)');
+	if (!byline) {
+		return;
+	}
+
+	byline.classList.add('rgh-clean-conversation-header');
+
+	const author = select('.author', byline)!;
+	const isSameAuthor = pageDetect.isPRConversation() && author.textContent === (await elementReady('.TimelineItem .author'))!.textContent;
+
+	const [base, headBranch] = select.all('.commit-ref', byline)!;
+	const baseBranch = base.title.split(':')[1];
 
-			// Removes: [octocat wants to merge 1] commit into github:master from octocat:feature
-			// Removes: [octocat] merged 1 commit into master from feature
-			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 2 : 4)) {
-				node.remove();
-			}
+	// Replace the word "from" with an arrow
+	headBranch.previousSibling!.replaceWith(<> <ArrowLeftIcon/> </>);
```

Comment:
I don't like leaving identical reviews 
```suggestion
	headBranch.previousSibling!.replaceWith(' ', <ArrowLeftIcon/>, ' ');
```

Reply:
Sorry I did not notice that part of the change.